### DISCUSSION TITLE
Why does RegisterUpdateListener fire several times for one update?

### DISCUSSION THREAD

**Discussion by 6peterlu** [facebook/lexical#2624] [Q&A]
I'm trying to trigger updates on selecting text with the following code block:

```
export default function CommentPlugin(): null {
    const [editor] = useLexicalComposerContext();
    editor.registerUpdateListener(({editorState, tags}) => {
        editorState.read(() => {
            const selection = $getSelection();
            if ($isRangeSelection(selection) && selection.anchor.offset !== selection.focus.offset) {
                console.log('comment plugin: range selection', selection);
            }
        });
    });
    return null;
}
```

When I highlight one character in the text, I see 8 log statements as follows:
<img width="476" alt="image" src="https://user-images.githubusercontent.com/8334251/178127049-7ddeee13-bb86-4c06-9c11-713b26a9c618.png">

There always seems to be 8 per update. Why might this be happening?

Thanks so much!

**Comment by trueadm** [comment]
Do you have any logic that triggers updates from with an update listener? It could be leading to a waterfall of updates (you can avoid this by using node transforms instead). However, given you're saying that you are trying to highlight a word, it could actually be correct. Each update might be the addition of a single character being added to the range of selection.

**Comment by 6peterlu** [comment]
I am using the following builtin plugins:

```
          <RichTextPlugin
            placeholder={<TextPlaceholder placeholderText='Enter some text...'/>}
            contentEditable={
              <ContentEditable
                style={{ outline: 0, flex: 1 }}
              />
            }
          />
          <AutoFocusPlugin />
          <ListPlugin />
          <MarkdownShortcutPlugin
            transformers={TRANSFORMERS}
          />
```

I'm not sure if any of these would trigger updates on select? I am getting 8 updates per character, not for the whole word. 

Thanks for the tip about node transforms, I will be sure to pay attention to that as I keep developing.
